src/config/mongoose_config_parser_toml.erl

```diff
@@ -1518,24 +1529,40 @@ handler([_, <<"sasl_mechanisms">>, <<"auth">>]) -> fun sasl_mechanism/2;
-handler([_, <<"connection">>, _, <<"redis">>, <<"outgoing_pools">>]) -> fun redis_option/2;
-handler([_, <<"connection">>, _, <<"ldap">>, <<"outgoing_pools">>]) -> fun ldap_option/2;
-handler([_, <<"servers">>, <<"connection">>, _, <<"ldap">>, <<"outgoing_pools">>]) -> fun(_, V) -> [b2l(V)] end;
-handler([_, <<"connection">>, _, <<"riak">>, <<"outgoing_pools">>]) -> fun riak_option/2;
-handler([_, <<"credentials">>, <<"connection">>, _, <<"riak">>, <<"outgoing_pools">>]) -> fun riak_credentials/2;
-handler([_, <<"connection">>, _, <<"cassandra">>, <<"outgoing_pools">>]) -> fun cassandra_option/2;
-handler([_, <<"servers">>, <<"connection">>, _, <<"cassandra">>, <<"outgoing_pools">>]) -> fun cassandra_server/2;
-handler([_, <<"connection">>, _, <<"elastic">>, <<"outgoing_pools">>]) -> fun elastic_option/2;
-handler([_, <<"connection">>, _, <<"rabbit">>, <<"outgoing_pools">>]) -> fun rabbit_option/2;
-handler([<<"keepalive_interval">>, <<"connection">>, _, <<"rdbms">>, <<"outgoing_pools">>]) -> 
-    fun(_, V) -> V end;
-handler([_, <<"tls">>, <<"connection">>, _, _, <<"outgoing_pools">>]) -> fun tls_option/2;
-handler([_, <<"versions">>, <<"tls">>, <<"connection">>, _, _, <<"outgoing_pools">>]) ->
+handler([_, {connection, odbc}, _,
+         <<"rdbms">>, <<"outgoing_pools">>]) -> fun odbc_option/2;
+handler([_, {connection, _}, _,
+         <<"rdbms">>, <<"outgoing_pools">>]) -> fun sql_server_option/2;
+handler([_, <<"connection">>, _,
+         <<"http">>, <<"outgoing_pools">>]) -> fun http_option/2;
+handler([_, <<"http_opts">>, <<"connection">>, _,
+         <<"http">>, <<"outgoing_pools">>]) -> fun http_opts/2;
+handler([_, <<"connection">>, _,
+         <<"redis">>, <<"outgoing_pools">>]) -> fun redis_option/2;
+handler([_, <<"connection">>, _,
+         <<"ldap">>, <<"outgoing_pools">>]) -> fun ldap_option/2;
+handler([_, <<"servers">>, <<"connection">>, _,
+         <<"ldap">>, <<"outgoing_pools">>]) -> fun(_, V) -> [b2l(V)] end;
+handler([_, <<"connection">>, _,
+         <<"riak">>, <<"outgoing_pools">>]) -> fun riak_option/2;
+handler([_, <<"credentials">>, <<"connection">>, _,
+         <<"riak">>, <<"outgoing_pools">>]) -> fun riak_credentials/2;
+handler([_, <<"connection">>, _,
+         <<"cassandra">>, <<"outgoing_pools">>]) -> fun cassandra_option/2;
+handler([_, <<"servers">>, <<"connection">>, _,
+         <<"cassandra">>, <<"outgoing_pools">>]) -> fun cassandra_server/2;
+handler([_, <<"connection">>, _,
+         <<"elastic">>, <<"outgoing_pools">>]) -> fun elastic_option/2;
+handler([_, <<"connection">>, _,
+         <<"rabbit">>, <<"outgoing_pools">>]) -> fun rabbit_option/2;
+handler([_, <<"tls">>, _, _, _, <<"outgoing_pools">>]) -> fun tls_option/2;
+handler([_, <<"versions">>, <<"tls">>, _, _, _, <<"outgoing_pools">>]) ->
     fun(_, Val) -> [b2a(Val)] end;
-handler([_, <<"ciphers">>, <<"tls">>, <<"connection">>, _, _, <<"outgoing_pools">>]) ->
```

Comment:
Can `tls` options be someplace different than in the `connection` section?

Reply:
Actually the only issue is that now we can have either `<<"connection">>` or `{connection, DBType}` there as the DB type is stored in the path to be able to have different handlers for different DB types. I thought that pattern-matching for both was a bit verbose and that there is not much risk with parsing something else as it is at the end of the list for `outgoing_pools`